src/app/clusters/basic/basic.cpp

```diff
@@ -62,41 +62,251 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(hardwareVersion);
+        }
+        break;
+    }
+
+    case HardwareVersionString::Id: {
+        constexpr size_t kMaxLen                = DeviceLayer::ConfigurationManager::kMaxHardwareVersionStringLength;
+        char hardwareVersionString[kMaxLen + 1] = { 0 };
+        status = ConfigurationMgr().GetHardwareVersionString(hardwareVersionString, sizeof(hardwareVersionString));
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(hardwareVersionString, strnlen(hardwareVersionString, kMaxLen)));
+        }
+        break;
+    }
+
+    case SoftwareVersion::Id: {
+        uint32_t softwareVersion = 0;
+        status                   = ConfigurationMgr().GetSoftwareVersion(softwareVersion);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(softwareVersion);
+        }
+        break;
+    }
+
+    case SoftwareVersionString::Id: {
+        constexpr size_t kMaxLen                = DeviceLayer::ConfigurationManager::kMaxSoftwareVersionStringLength;
+        char softwareVersionString[kMaxLen + 1] = { 0 };
+        status = ConfigurationMgr().GetSoftwareVersionString(softwareVersionString, sizeof(softwareVersionString));
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(softwareVersionString, strnlen(softwareVersionString, kMaxLen)));
+        }
+        break;
+    }
+
+    case ManufacturingDate::Id: {
+        constexpr size_t kMaxLen                  = DeviceLayer::ConfigurationManager::kMaxManufacturingDateLength;
+        char manufacturingDateString[kMaxLen + 1] = { 0 };
+        uint16_t manufacturingYear;
+        uint8_t manufacturingMonth;
+        uint8_t manufacturingDayOfMonth;
+        status = ConfigurationMgr().GetManufacturingDate(manufacturingYear, manufacturingMonth, manufacturingDayOfMonth);
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+        {
+            manufacturingYear       = 2020;
+            manufacturingMonth      = 1;
+            manufacturingDayOfMonth = 1;
+            status                  = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            // Format is YYYYMMDD
+            snprintf(manufacturingDateString, sizeof(manufacturingDateString), "%04" PRIu16 "%02" PRIu8 "%02" PRIu8,
+                     manufacturingYear, manufacturingMonth, manufacturingDayOfMonth);
+            status = aEncoder.Encode(chip::CharSpan(manufacturingDateString, strnlen(manufacturingDateString, kMaxLen)));
+        }
+        ChipLogError(Zcl, "MAAAAANNNNUFFF: %s, err: %" CHIP_ERROR_FORMAT, manufacturingDateString, status.Format());
```

Comment:
I don't think we want the unconditional ChipLogError here.

Reply:
Sorry, was debuggign sometjing from CI failure and this got left behind.